### ISSUE TITLE
Request: Add Modifier "Weighted Normal"

### ISSUE THREAD

**Issue by torridgristle** [jayanam/jmesh-tools#2]
The Weighted Normal modifier is basically made for this, non-subdivided geometry with ngons and bevels. Needs the line `bpy.ops.object.modifier_add(type='WEIGHTED_NORMAL')` added after the code for adding the bevel modifier in fc_bevel_util.py and nothing else; the default values seem to be just fine.

**Comment by jayanam** [comment]
Please explain the use cases a bit more when this is needed

**Comment by torridgristle** [comment]
Normally the normals for the edges of faces will either be blended between the larger face, which we would want to be flat, and the neighboring faces of the bevel; or it will be split off entirely due to auto-smooth detecting too extreme of an angle in which case the shading of the bevel's faces don't blend into the larger faces. The weighted normals modifier forces to the larger surfaces to always get priority when blending between normals for smooth shading so that the larger faces' normals are always facing directly outward and the normals of the bevels adjust to match at the edges of these larger faces. [This image describes a very extreme situation where the bevel is only one face but the concept is the same.](https://i.imgur.com/ikZjE2B.png)

tl;dr it's like autosmooth but without the discontinuity in surface smoothing at bevel edges due to split edges.

**Comment by torridgristle** [comment]
Autosmooth (No Sharp Edges)
https://i.imgur.com/bW15IOs.png

Weighted Normals
https://i.imgur.com/ikPY6Qu.png

Autosmooth (Sharp Edges)
https://i.imgur.com/DdatP5C.png

Weighted Normals
https://i.imgur.com/sNOBeuM.png

**Comment by jayanam** [comment]
Added the weighted normal after bevel and also remove it on un-bevel
